packages/pilot/src/pages/ErrorBoundary/reducer.js

```diff
@@ -0,0 +1,224 @@
+  equals(['apiError', true])
+)
+
+const removeErrorsByStatusCode = uncurryN(
+  2,
+  codes => reject(either(
+    hasStatusInCodes(codes),
+    isPrioritaryError
+  ))
+)
+
+const passAnyRoute = uncurryN(2, route => pipe(
+  map(item => item.test(route)),
+  any(equals(true))
+))
+
+const passExactRoute = uncurryN(2, route => pipe(
+  map(pipe(
+    regex => regex.exec(route),
+    defaultTo([]),
+    head,
+    equals(route)
+  )),
+  any(equals(true))
+))
+
+const regexTest = uncurryN(2, propName => pipe(
+  prop(propName),
+  ifElse(
+    is(RegExp),
+    test,
+    equals
+  )
+))
+
+const matchesRegex = uncurryN(2, regex => evolve({
+  message: regexTest('message', regex),
+  method: regexTest('method', regex),
+  name: regexTest('name', regex),
+  status: regexTest('status', regex),
+  type: regexTest('type', regex),
+}))
+
+const matchesApiError = (error, localError) => {
+  const hasRoute = is(Array, localError.affectedRoutes)
+    ? passAnyRoute(error.affectedRoute, localError.affectedRoutes)
+    : true
+
+  const {
+    message,
+    method,
+    name,
+    status,
+    type,
+  } = matchesRegex(localError, error)
+
+  return hasRoute
+    && message
+    && method
+    && name
+    && status
+    && type
+}
+
+const matchesReactError = (error, { affectedRoutes, message, name }) => {
+  const hasRoute = is(Array, affectedRoutes)
+    ? passAnyRoute(error.affectedRoute, affectedRoutes)
+    : true
+
+  const matchesMessage = is(RegExp, message) && message.test(error.message)
+  const matchesName = is(RegExp, name) && name.test(error.name)
+
+  return hasRoute && matchesMessage && matchesName
+}
+
+const findError = (error, errors) => {
+  const sameRouteErrors = errors.filter(({ affectedRoutes }) =>
+    passExactRoute(error.affectedRoute, affectedRoutes))
+
+  const differentRouteErrors = difference(errors, sameRouteErrors)
+  const sameRouteErrorsFirst = concat(sameRouteErrors, differentRouteErrors)
+
+  const foundError = sameRouteErrorsFirst.find((localError) => {
+    let result = false
+
+    if (error.source === 'api') {
+      result = matchesApiError(error, localError)
+    } else if (error.source === 'react') {
+      result = matchesReactError(error, localError)
+    }
+
+    return result
+  })
+
+  return foundError
+}
+
+const addErrorComponent = (error) => {
+  const localError = findError(error, localErrors)
+
+  if (localError) {
+    const { getComponent } = localError
+
+    const localized = error.source === 'react'
+      ? t => ({
+        message: t(error.message),
+        title: t('pages.error.application_title'),
+      })
+      : localError.localized
+
+    return {
+      ...error,
+      getComponent,
+      localized,
+      priority: !!getComponent,
+    }
+  }
+
+  return error
+}
+
+const addError = (error, state) => {
+  const errorWithComponent = addErrorComponent(error)
+  return addNewError(errorWithComponent, state)
+}
+
+export default function loginReducer (state = initialState, action) {
```

Comment:
```suggestion
export default function errorReducer (state = initialState, action) {
```
errorReducer parece fazer mais sentido nesse contexto